### PR TITLE
Ros calibration support for stereo cameras. 

### PR DESCRIPTION
**Pull request by v-mehta**
- Import script for stereo images from ROS+OpenCV calibration.

- Config and helper scripts for Kalibr data input and output.

- Launch files for using the Kalibr camchain params in usual ros stack with Clearpath pipelines.

### CODE REVIEW

File: aslam_offline_calibration/kalibr/config/target.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+target_type: 'checkerboard' #gridtype
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/config/target.yaml]: pedantic: space after `#`

File: aslam_offline_calibration/kalibr/launch/undistort_kalibr.launch [hunk 2]
```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0"?>
+
+<launch>
+  <!-- This argument needs to be set for importing the params. -->
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/launch/undistort_kalibr.launch:4]: optional: you can add `doc` attributes to the `arg` elements, and put the comments/description there.

Comment (v-mehta) [reply aslam_offline_calibration/kalibr/launch/undistort_kalibr.launch:4]: Great idea!!

**Review by servos — APPROVED** [review]

File: aslam_offline_calibration/kalibr/python/kalibr_calibrate_cameras [hunk 3]
```diff
@@ -389,54 +390,54 @@ def main():
...
             if parsed.verbose and len(calibrator.baselines)>1:
 		        f=pl.figure(100006)
 		        kcc.plotCameraRig(calibrator.baselines, fno=f.number, clearFigure=False)
 		        pl.show()
-            
+
             #write to file
             bagtag = parsed.bagfile.translate(None, "<>:/\|?*").replace('.bag', '', 1)
-            resultFile = "camchain-" + bagtag + ".yaml"
+            resultFile = parsed.output_folder + "/" + "camchain" + ".yaml"
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/python/kalibr_calibrate_cameras]: `resultFile = os.path.join(parsed.output_folder, "camchain" + ".yaml")`

and same for nearby lines

**Review by efernandez — CHANGES_REQUESTED** [review]

File: aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+for file in left-*.png; do
+  mv "$file" "./cam0/${file/left-/}"
+done
+
+mkdir cam1
+for file in right-*.png; do
+  mv "$file" "./cam1/${file/right-/}"
+done
+
+cd -
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh]: This won't return to the original `pwd` (before calling the script). You need to save it on a variable and `cd` to it.

``` bash
OLDPWD=`pwd`

# bunch of code

cd "$OLDPWD"
```

File: aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+folder=$1
+cd $folder
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh]: I'd add some sanity checks here: check the folder exists, the tar file too, etc.

File: aslam_offline_calibration/kalibr/python/yaml_to_camera_info_publisher [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    camera_info_msg.width = calib_data["image_width"]
+    camera_info_msg.height = calib_data["image_height"]
+    camera_info_msg.K = calib_data["camera_matrix"]["data"]
+    camera_info_msg.D = calib_data["distortion_coefficients"]["data"]
+    camera_info_msg.R = calib_data["rectification_matrix"]["data"]
+    camera_info_msg.P = calib_data["projection_matrix"]["data"]
+    camera_info_msg.distortion_model = calib_data["distortion_model"]
+    camera_info_msg.binning_x = 1
+    camera_info_msg.binning_y = 1
+    # camera_info_msg.
```
Comment (efernandez) [inline aslam_offline_calibration/kalibr/python/yaml_to_camera_info_publisher]: leftover ? -> remove

**Review by efernandez — CHANGES_REQUESTED** [review]

File: aslam_offline_calibration/kalibr/python/yaml_to_camera_info_publisher (hunk 6 above)
Comment (v-mehta) [reply aslam_offline_calibration/kalibr/python/yaml_to_camera_info_publisher]: Yeah. I was comparing ROS+OpenCV yaml to camera info against Kalibr Camchain params. Do not need this script anymore.

File: aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh (hunk 4 above)
Comment (v-mehta) [reply aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh]: Done.

File: aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh (hunk 5 above)
Comment (v-mehta) [reply aslam_offline_calibration/kalibr/scripts/kalibr_data_import.sh]: Good point. Added checks for the folder and tar file.

File: aslam_offline_calibration/kalibr/python/kalibr_calibrate_cameras (hunk 3 above)
Comment (v-mehta) [reply aslam_offline_calibration/kalibr/python/kalibr_calibrate_cameras]: Done.

File: aslam_offline_calibration/kalibr/config/target.yaml (hunk 1 above)
Comment (v-mehta) [reply aslam_offline_calibration/kalibr/config/target.yaml]: Done.

**Review by efernandez — APPROVED** [review]

**Comment by efernandez** [comment]
Thanks! :+1:

**Pull request merged**